detection_rules/rule.py

```diff
@@ -632,7 +641,7 @@ class ThresholdQueryRuleData(QueryRuleData):
     class ThresholdMapping(MarshmallowDataclassMixin):
         @dataclass(frozen=True)
         class ThresholdCardinality:
-            field: str
+            field: Optional[Union[str, List[str]]]
```

Comment:
Whats up with this change?

Reply:
if the structure change along the way, we need to maintain backwards compat

Reply:
Good catch. I mixed this up with the base field below when I looked at the kibana implementation.